truss/truss_handle.py

```diff
@@ -397,12 +397,37 @@ def training_docker_build_setup(self, build_dir: Optional[Path] = None):
 
     def add_python_requirement(self, python_requirement: str):
         """Add a python requirement to truss model's config."""
-        self._update_config(
-            lambda conf: replace(
-                conf, requirements=[*conf.requirements, python_requirement]
-            )
+
+        # Parse the added python requirements
+        parsed_reqs = pkg_resources.Requirement.parse(python_requirement)
+        requirement_name = parsed_reqs.name
+        _, requirement_version = parsed_reqs.specs[0]
+
+        # Parse the existing requirements
+        requirements = list(
+            pkg_resources.parse_requirements(self._spec.config.requirements)
         )
 
+        # If we already have the requirement among the existing requirements
+        # but version needs to be updated
+        if requirement_name in map(lambda x: x.name, requirements):
+            [
+                list(req.specs[0])[1].replace(
+                    list(req.specs[0])[1], requirement_version
+                )
+                for req in requirements
+                if req.name == requirement_name
+            ]
+            self._update_config(lambda conf: replace(conf, requirements=[]))
```

Comment:
This reads quite complex, how about something like this:
```python
    def add_python_requirement(self, python_requirement: str):
        """Add a python requirement to truss model's config."""

        input_python_req_name = _python_req_name(python_requirement)
        new_reqs = [req for req in self._spec.config.requirements 
                              if _python_req_name(req) != input_python_req_name]
        new_reqs.append(python_requirement)
        self._update_config(
            lambda conf: replace(conf,  requirements=[new_reqs])
        )


    def _python_req_name(python_requirement: str) -> str:
        return pkg_resources.Requirement.parse(python_requirement)


```

Reply:
I worked on improving the readability of the code based on the suggestions, thank you for that, hope it looks better now